packages/db/src/connection.ts

```diff
@@ -0,0 +1,129 @@
+import Knex from 'knex';
+import { getSecretConnectionConfig, getEnvConnectionConfig } from './config';
+
+import { envConectionConfigObject, knexSecretConnectionConfigObject } from './types';
+
+export interface knexConnectionConfigObject {
+  host: string,
+  username: string,
+  password: string,
+  database?: string,
+}
+
+interface knexConfigObject {
+  client: string,
+  connection: knexConnectionConfigObject,
+  acquireConnectionTimeout: number,
+  asyncStackTraces: boolean,
+  debug: boolean,
+  migrations?: {
+    directory: string
+  }
+}
+
+interface knexEnvironmentObject extends NodeJS.ProcessEnv {
+  KNEX_DEBUG?: string
+  KNEX_ASYNC_STACK_TRACES?: string,
+}
+
+/**
+* @summary Given a knexEnvironmentObject and a Knex.PgConnectionConfig, returns
+* a Knex instance with those confiugrations applied
+*
+* @param {Knex.PgConnectionConfig} connectionConfig - Knex connection configuration
+* @param {knexEnvironmentObject} env - environment object with knex configuration keys
+* @returns {Knex} - configured Knex instance
+*/
+const getConfiguredKnex = (
+  connectionConfig: Knex.PgConnectionConfig,
+  env: knexEnvironmentObject
+): Knex => {
+  let knexConfig = {
+    client: 'pg',
+    connection: connectionConfig,
+    debug: env?.KNEX_DEBUG === 'true',
+    asyncStackTraces: env?.KNEX_ASYNC_STACK_TRACES === 'true',
+    acquireConnectionTimeout: env?.timeout ? env.timeout : 60000,
+  } as knexConfigObject;
+
+  if (env?.migrationDir) {
```

Comment:
If you update line 41 to use the `Knex.Config` type, then this can be simplified to:

```js
if (env.migrationDir) {
  knexConfig.migrations = { directory: env.migrationDir };
}
```